llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -883,6 +881,110 @@ static bool removeDeviceGlobalFromCompilerUsed(Module &M) {
   return true;
 }
 
+// Add extra red zone to each image scope device globals if the module has been
+// instrumented by sanitizer pass. And record their infomation like size, red
+// zone size, beginning address.
+static bool instrumentDeviceGlobal(Module &M) {
```

Comment:
I think it should be a Pass rather than a simple function.

Reply:
Hi @maksimsab, we just do this instrumentation in sycl-post-link temporarily. Once UR PR https://github.com/oneapi-src/unified-runtime/pull/1255 merged, I'll move this code back to AddressSanitizer pass. If you still think I need to write a pass to this, it's also ok for me. Do you think it's necessary to write a pass?

Reply:
Done, write a pass for this.